add modules 'bank_statement_import_invoices_fix' and 'bank_statement_hide_bank_in_lines'

damdam-s opened the pull request:
'bank_statement_hide_bank_in_lines' :: This module was written to add a group to be able to hide/show the field 'bank_account_id' in the bank statement lines

'bank_statement_import_invoices_fix' :: This module was written to extend the functionality of importing invoices as bank statement lines. It invert values of "communication" and "reference" fields

damdam-s committed:
[ADD] add module 'bank_statement_import_invoices_fix'

damdam-s committed:
[ADD] add module 'bank_statement_hide_bank_in_lines'

yvaucher commented on bank_statement_hide_bank_in_lines/README.rst:
```diff
@@ -0,0 +1,70 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+    :alt: License: AGPL-3
+
+Module name
```
To replace by name of the module as title

yvaucher commented on bank_statement_hide_bank_in_lines/__openerp__.py:
```diff
@@ -0,0 +1,35 @@
+    'installable': True,
+    'auto_install': False,
+}
+# vim:expandtab:smartindent:tabstop=4:softtabstop=4:shiftwidth=4:
```
No need to put vim line

yvaucher commented on bank_statement_hide_bank_in_lines/models/models.py:
```diff
@@ -0,0 +1,20 @@
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
```
As @sbidoul said, please remove this file

yvaucher commented on bank_statement_hide_bank_in_lines/security/account_security.xml:
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="utf-8"?>
+<openerp>
+	<data noupdate="0">
```
Don't use <tab> for XML indent. 2 spaces or 4 spaces are correct.

yvaucher commented on bank_statement_hide_bank_in_lines/views/bank_statement_hide_bank_in_lines_view.xml:
```diff
@@ -0,0 +1,19 @@
+            </field>
+        </record>
+
+	
```
Extra line with tab

yvaucher commented on bank_statement_import_invoices_fix/README.rst:
```diff
@@ -0,0 +1,71 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+    :alt: License: AGPL-3
+
+Module name
```
Put the name of the module here.

yvaucher commented on bank_statement_import_invoices_fix/__openerp__.py:
```diff
@@ -0,0 +1,33 @@
+    'installable': True,
+    'auto_install': False,
+}
+# vim:expandtab:smartindent:tabstop=4:softtabstop=4:shiftwidth=4:
```
vim line to remove

yvaucher commented on bank_statement_import_invoices_fix/models/models.py:
```diff
@@ -0,0 +1,22 @@
+#
+##############################################################################
+
+# from openerp import models, fields, api
```
This file should be removed

yvaucher commented on bank_statement_import_invoices_fix/wizard/account_statement_from_invoice.py:
```diff
@@ -0,0 +1,47 @@
+
+        return {'type': 'ir.actions.act_window_close'}
+
+# vim:expandtab:smartindent:tabstop=4:softtabstop=4:shiftwidth=4:
```
vim line to remove

yvaucher commented on bank_statement_import_invoices_fix/wizard/account_statement_from_invoice.py:
```diff
@@ -0,0 +1,47 @@
+        if statement_id:
+            statement = bank_statement_obj.browse(statement_id)
+            for line in statement.line_ids:
+                line.write({'name': line.ref, 'ref': line.name})
```
This also requires https://github.com/odoo/odoo/pull/7608 to be merged upstream.

By the way, https://github.com/OCA/bank-payment/pull/187 offer the same inversion when creating the lines, so installing both modules won't work as it would invert it twice.

yvaucher commented on bank_statement_import_invoices_fix/wizard/account_statement_from_invoice.py:
```diff
@@ -0,0 +1,47 @@
+        """
+        Inverse data from account_move_line
+        bank_statement_line.name = line.ref
+        bank_statement_line.ref = name.name
```
line.name ?

yvaucher commented on bank_statement_import_invoices_fix/wizard/account_statement_from_invoice.py:
```diff
@@ -0,0 +1,47 @@
+        """
+        super(account_statement_from_invoice_lines, self).populate_statement()
+        bank_statement_obj = self.env['account.bank.statement']
+        statement_id = self._context.get('statement_id', False)
```
False is not required, get will already return None by default

yvaucher commented on bank_statement_import_invoices_fix/__openerp__.py:
```diff
@@ -0,0 +1,33 @@
+#
+##############################################################################
+{
+    "name": "Bank statement import invoices fix",
```
Can you add a summary ?

yvaucher commented on bank_statement_import_invoices_fix/README.rst:
```diff
@@ -0,0 +1,71 @@
+* https://www.odoo.com/forum/help-1
+
+Known issues / Roadmap
+======================
```
Please remove this empty section

yvaucher commented:
I fear that module bank_statement_import_invoices_fix will conflict with module account_import_line_multicurrency_extension as both are switching the ref and name fields.

The second one might need to depends on the first one.

yvaucher commented on bank_statement_import_invoices_fix/README.rst:
```diff
@@ -0,0 +1,71 @@
+
+This module was written to extend the functionality of importing invoices as bank statement lines.
+It invert values of "communication" and "reference" fields
+
```
This sentence could help to explain why:

When lines are imported the reference goes to the communication field (instead of reference of the bank statement line) in order to have a default proposal from the system during the reconciliation process.

(from module account_import_line_multicurrency_extension which was doing the same)

yvaucher commented on bank_statement_hide_bank_in_lines/README.rst:
```diff
@@ -0,0 +1,70 @@
+* https://www.odoo.com/forum/help-1
+
+Known issues / Roadmap
+======================
```
You can remove this empty section

yvaucher commented on bank_statement_hide_bank_in_lines/README.rst:
```diff
@@ -0,0 +1,70 @@
+Bugs are tracked on `GitHub Issues <https://github.com/OCA/{project_repo}/issues>`_.
+In case of trouble, please check there if your issue has already been reported.
+If you spotted it first, help us smashing it by providing a detailed and welcomed feedback
+`here <https://github.com/OCA/{project_repo}/issues/new?body=module:%20{module_name}%0Aversion:%20{version}%0A%0A**Steps%20to%20reproduce**%0A-%20...%0A%0A**Current%20behavior**%0A%0A**Expected%20behavior**>`_.
```
Please fill the placeholders

yvaucher commented on bank_statement_hide_bank_in_lines/README.rst:
```diff
@@ -0,0 +1,70 @@
+Bug Tracker
+===========
+
+Bugs are tracked on `GitHub Issues <https://github.com/OCA/{project_repo}/issues>`_.
```
Fill the placeholder

yvaucher commented on bank_statement_import_invoices_fix/models/__init__.py:
```diff
@@ -0,0 +1,20 @@
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
```
File to remove

yvaucher commented:
To squash that way:

```
pick 22b6de2 [ADD] add module 'bank_statement_import_invoices_fix'
fixup dd3ed1c [FIX] fix flake8
fixup 3d06fb1 [DEL] remove eclipse config file
pick a5840e3 [ADD] add module 'bank_statement_hide_bank_in_lines'
fixup a1e125d [FIX] fix flake8
```

yvaucher commented:
For eclipse config file add it in a global gitignore file

yvaucher committed:
Remove empty files

yvaucher committed:
Fix xml indent

yvaucher committed:
Remove vimlines

yvaucher committed:
Improve README.rst

Setting placeholders, extending module description, removing empty
sections and giving the module name to the title

for modules:
- bank_statement_hide_bank_in_lines
- bank_statement_import_invoices_fix

yvaucher commented:
Waiting to push force changes from https://github.com/yvaucher/bank-statement-reconcile/tree/8.0-dams-bank-statement_import-fixes to replace this branch

Did the fixup and fixed other points

yvaucher committed:
Add summary

yvaucher committed:
[TYPO]

yvaucher committed:
Remove default value of get to False as it is None if not specified

pedrobaeza commented:
Closing as CIs statuses are red and very old. Feel free to reopen if still interested.
